src/store/utils/queryParams.ts

```diff
@@ -2,7 +2,9 @@ import Router from 'next/router';
 
 export const setQueryParameter = (key: string, value: string) => {
   if (typeof window !== 'undefined') {
-    Router.push({ query: { [key]: value } }, undefined, { shallow: true });
+    Router.push({ query: { ...getAllQueryParameters(), [key]: value } }, undefined, {
```

Comment:
as this runs on app startup the Router.state.query might not be ready yet so using native apis to fetch current parameters

Reply:
You left a `console.log` in